stacker/blueprints/base.py

```diff
@@ -216,7 +219,55 @@ def resolve_variable(var_name, var_def, provided_variable, blueprint_name):
     return value
 
 
+def parse_user_data(variables, raw_user_data, blueprint_name):
+    """Parse the given user data and renders it as a template
+
+    It supports referencing template variables to create userdata
+    that's supplemented with information from the stack, as commonly
+    required when creating EC2 userdata files.
+
+    Args:
+        variables (dict): variables available to the template
+        raw_user_data (str): the user_data to be parsed
+        blueprint_name (str): the name of the blueprint
+
+    Returns:
+        str: The parsed user data, with all the variables values and
+             refs replaced with their resolved values.
+
+    Raises:
+        MissingVariable: Raised when a variable is in the user_data that
+                         is not given in the blueprint
+
+        InvalidUserdataPlaceholder: Raised when a placeholder name in user_data
+                                    is not valid. E.g ${100} would raise this.
+
+
+        """
```

Comment:
This indentation is still off, should be flush with the start of the doc string.